sqlite/func.c

```diff
@@ -1155,6 +1156,53 @@ static void nullifFunc(
   }
 }
 
+static void shaFunc(
+  sqlite3_context *context,
+  int argc,
+  sqlite3_value **argv
+){
+  const unsigned char * data = sqlite3_value_text(argv[0]);
+  int len = sqlite3_value_bytes(argv[0]);
+
+  assert( argc==1 );
+  UNUSED_PARAMETER(argc);
+
+  unsigned char hash[SHA256_DIGEST_LENGTH];
+  char hashStr[SHA256_DIGEST_LENGTH * 2];
+
+  switch( sqlite3_value_type(argv[0]) ){
+    case SQLITE_INTERVAL_YM:
+    case SQLITE_INTERVAL_DS:
+    case SQLITE_INTERVAL_DSUS:
+    case SQLITE_DATETIME:
+    case SQLITE_DATETIMEUS:
+    case SQLITE_DECIMAL:
+    case SQLITE_INTEGER:
+    case SQLITE_FLOAT:
+    case SQLITE_BLOB:
+    case SQLITE_TEXT: {
+      SHA256_CTX sha256;
+      SHA256_Init(&sha256);
+
+      SHA256_Update(&sha256, data, len);
+      SHA256_Final(hash, &sha256);
```

Comment:
These 3 functions return an `int` return value, which should be checked.

Or use the `SHA256()` convenience function.